thermalnetwork/network.py

```diff
@@ -579,10 +553,42 @@ def size(self, output_path: Path, throw: bool = True):
         elif self.des_method == DesignType.UPSTREAM:
             self.size_to_upstream_equipment(output_path)
         else:
-            if throw:
-                logger.error(f"Unsupported design method {self.des_method}")
-            return 1
-        return 0
+            msg = f"Unsupported design method {self.des_method}"
+            logger.error(msg)
+            raise ValueError(msg)
+
+    def size_network(self, system_parameter_path: Path):
```

Comment:
I'm not going to go line by line on the rest. It's basically an massive refactor of a lot of things, some for general workflow cleanup, and some to accommodate the new schema and methods. Tag me in a comment if you have a question about something specific.